[2024-11-08T18:16:32.939+0000] {local_task_job_runner.py:123} INFO - ::group::Pre task execution logs
[2024-11-08T18:16:33.087+0000] {taskinstance.py:2612} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: fraud_detection.process_transaction scheduled__2024-11-08T18:15:00+00:00 [queued]>
[2024-11-08T18:16:33.112+0000] {taskinstance.py:2612} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: fraud_detection.process_transaction scheduled__2024-11-08T18:15:00+00:00 [queued]>
[2024-11-08T18:16:33.113+0000] {taskinstance.py:2865} INFO - Starting attempt 1 of 2
[2024-11-08T18:16:33.151+0000] {taskinstance.py:2888} INFO - Executing <Task(BranchPythonOperator): process_transaction> on 2024-11-08 18:15:00+00:00
[2024-11-08T18:16:33.170+0000] {warnings.py:112} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:70: DeprecationWarning: This process (pid=934) is multi-threaded, use of fork() may lead to deadlocks in the child.
  pid = os.fork()

[2024-11-08T18:16:33.172+0000] {standard_task_runner.py:72} INFO - Started process 938 to run task
[2024-11-08T18:16:33.177+0000] {standard_task_runner.py:104} INFO - Running: ['***', 'tasks', 'run', 'fraud_detection', 'process_transaction', 'scheduled__2024-11-08T18:15:00+00:00', '--job-id', '211', '--raw', '--subdir', 'DAGS_FOLDER/fraud_detection_dag.py', '--cfg-path', '/tmp/tmp2eb6_iuy']
[2024-11-08T18:16:33.180+0000] {standard_task_runner.py:105} INFO - Job 211: Subtask process_transaction
[2024-11-08T18:16:33.347+0000] {task_command.py:467} INFO - Running <TaskInstance: fraud_detection.process_transaction scheduled__2024-11-08T18:15:00+00:00 [running]> on host 641085defe8e
[2024-11-08T18:16:33.707+0000] {taskinstance.py:3131} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='fraud_team' AIRFLOW_CTX_DAG_ID='fraud_detection' AIRFLOW_CTX_TASK_ID='process_transaction' AIRFLOW_CTX_EXECUTION_DATE='2024-11-08T18:15:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2024-11-08T18:15:00+00:00'
[2024-11-08T18:16:33.708+0000] {taskinstance.py:731} INFO - ::endgroup::
[2024-11-08T18:16:33.726+0000] {baseoperator.py:405} WARNING - BranchPythonOperator.execute cannot be called outside TaskInstance!
[2024-11-08T18:16:33.825+0000] {fraud_detection_dag.py:77} INFO - Données récupérées depuis XCom: {"columns":["cc_num","merchant","category","amt","first","last","gender","street","city","state","zip","lat","long","city_pop","job","dob","trans_num","merch_lat","merch_long","is_fraud","current_time"],"index":[167011],"data":[[4777065439639721,"fraud_Dare-Marvin","entertainment",70.58,"Peter","Caldwell","M","08966 Beltran Route Suite 905","Oaks","PA",19456,40.1334,-75.4536,737,"Oceanographer","1987-04-24","de1f68e01805f7bbdb2ee225002483bc",40.504339,-76.387325,0,1731089789485]]}
[2024-11-08T18:16:33.827+0000] {fraud_detection_dag.py:85} INFO - Données JSON parsées avec succès
[2024-11-08T18:16:33.831+0000] {fraud_detection_dag.py:89} INFO - DataFrame créé avec 1 lignes
[2024-11-08T18:16:33.855+0000] {fraud_detection_dag.py:93} INFO - Colonnes renommées avec succès
[2024-11-08T18:16:33.857+0000] {fraud_detection_dag.py:94} INFO - Colonnes disponibles: ['cc_num', 'merchant', 'category', 'amt', 'first', 'last', 'gender', 'street', 'city', 'state', 'zip', 'lat', 'long', 'city_pop', 'job', 'dob', 'trans_num', 'merch_lat', 'merch_long', 'is_fraud', 'trans_date_trans_time']
[2024-11-08T18:16:33.861+0000] {fraud_detection_dag.py:98} INFO - Conversion de trans_date_trans_time en unix_time réussie
[2024-11-08T18:16:34.000+0000] {fraud_detection_dag.py:107} INFO - Transformation des données...
[2024-11-08T18:16:35.245+0000] {warnings.py:112} WARNING - /home/***/.local/lib/python3.12/site-packages/sklearn/ensemble/_base.py:172: UserWarning: Loky-backed parallel loops cannot be called in a multiprocessing, setting n_jobs=1
  n_jobs = min(effective_n_jobs(n_jobs), n_estimators)

[2024-11-08T18:16:35.260+0000] {warnings.py:112} WARNING - /home/***/.local/lib/python3.12/site-packages/sklearn/ensemble/_base.py:172: UserWarning: Loky-backed parallel loops cannot be called in a multiprocessing, setting n_jobs=1
  n_jobs = min(effective_n_jobs(n_jobs), n_estimators)

[2024-11-08T18:16:35.265+0000] {fraud_detection_dag.py:119} INFO - Prediction de fraude: NORMAL, Probabilité: 34.41%
[2024-11-08T18:16:35.304+0000] {fraud_detection_dag.py:125} INFO - Prédiction: NORMAL (Probabilité: 34.41%)
[2024-11-08T18:16:35.305+0000] {python.py:240} INFO - Done. Returned value was: notify_normal
[2024-11-08T18:16:35.306+0000] {branch.py:38} INFO - Branch into notify_normal
[2024-11-08T18:16:35.308+0000] {skipmixin.py:230} INFO - Following branch ('notify_normal',)
[2024-11-08T18:16:35.327+0000] {skipmixin.py:278} INFO - Skipping tasks [('skip_processing', -1), ('notify_fraud', -1)]
[2024-11-08T18:16:35.375+0000] {taskinstance.py:340} INFO - ::group::Post task execution logs
[2024-11-08T18:16:35.376+0000] {taskinstance.py:352} INFO - Marking task as SUCCESS. dag_id=fraud_detection, task_id=process_transaction, run_id=scheduled__2024-11-08T18:15:00+00:00, execution_date=20241108T181500, start_date=20241108T181633, end_date=20241108T181635
[2024-11-08T18:16:35.452+0000] {local_task_job_runner.py:266} INFO - Task exited with return code 0
[2024-11-08T18:16:35.506+0000] {taskinstance.py:3900} INFO - 1 downstream tasks scheduled from follow-on schedule check
[2024-11-08T18:16:35.514+0000] {local_task_job_runner.py:245} INFO - ::endgroup::
